C++ wx assertion errors pop up randomly during analysis runs

I've encountered this once or twice myself, but here are [some](http://forum.cellprofiler.org/t/cellprofiler-3-windows-10-large-dataset/5521/3) from [the](http://forum.cellprofiler.org/t/erro-message-c-assertion/5491) forum [recently](http://forum.cellprofiler.org/t/c-assertion-error-cp-3-0/5554); in only case did someone capture things from the terminal after (bottom snippet) but I've transcribed as best I can in the top snippet.  It's popped up in Windows 7, Mac, and Windows 10; the common thread seems to be that the pipelines run fine on smaller batches but hit this error on larger ones.  For at least 2 users running the pipeline with the display enabled ("eyes open") vs closed seemed to make a difference, one user said it did not and he had to decrease his batch size.  

I hadn't reported it before because it's an intermittent issue that apparently only crops up on large batch sizes, which I assumed would make it hard to debug and troubleshoot, but I realized it should probably be documented and hopefully you guys can do your magic and find it. 

![image](https://user-images.githubusercontent.com/6721515/35076303-b8c7435a-fbc5-11e7-9f80-c4b82d3424f8.png)
![image](https://user-images.githubusercontent.com/6721515/35076320-c8b7e2f6-fbc5-11e7-83ea-5b7d580b14aa.png)
![image](https://user-images.githubusercontent.com/6721515/35076293-a947217a-fbc5-11e7-82e6-1cf5b270028b.png)

```
C++ assertion "(itemid >=0 && itemid <SHRT_MAX) ||  (itemid>=wxID_AUTO_LOWEST && itemid <= wxID_AUTO_HIGHEST)" 
failed at ..\..\src\common\menucmn.cpp(260) in wxMenuItemBase::wxMenuItemBase(): invalid item value
```

```
Traceback (most recent call last):
File “CellProfiler\cellprofiler\gui\cpframe.py”, line 483, in OnClose
File “CellProfiler\cellprofiler\gui\pipelinecontroller.py”, line 1333, in check_close
File “site-packages\wx-3.0-msw\wx_controls.py”, line 193, in init
wx._core.PyAssertionError: C++ assertion “id == wxID_ANY || (id >= 0 && id < 32767) || (id >= wxID_AUTO_LOWEST && id <= wxID_AUTO_HIGHEST)” failed at …\src\common\wincmn.cpp(372) in wxWindowBase::CreateBase(): invalid id value
```

--- a/cellprofiler/gui/namesubscriber.py
+++ b/cellprofiler/gui/namesubscriber.py
@@ -42,7 +42,6 @@ class NameSubscriberComboBox(wx.Panel):
         if choices is None:
             choices = []
         self.orig_choices = choices
-        self.IDs = [wx.NewId() for c in choices]
         sizer = wx.BoxSizer(wx.HORIZONTAL)
         self.combo_dlg = wx.ComboBox(
                 self, choices=[choice[0] for choice in choices],
@@ -65,7 +64,6 @@ class NameSubscriberComboBox(wx.Panel):
         for child in self.combo_dlg.Children:
             # Mac implements read_only combobox as a choice in a child
             child.Bind(wx.EVT_RIGHT_DOWN, self.right_menu)
-        self.Bind(wx.EVT_MENU, self.menu_selected)
         self.callbacks = []
 
     def choice_made(self, evt):
@@ -97,12 +95,8 @@ class NameSubscriberComboBox(wx.Panel):
         menu = wx.Menu()
         all_menu = wx.Menu()
 
-        unsorted_choices = [
-            (name, annotation, num, is_input_module, identifier)
-            for (name, annotation, num, is_input_module), identifier in
-            zip(self.orig_choices, self.IDs)]
         fn_key = lambda x: (x[2], x)
-        choices_sorted_by_num = sorted(unsorted_choices, key=fn_key)
+        choices_sorted_by_num = sorted(self.orig_choices, key=fn_key)
         for name, annotation, num, is_input_module, choiceid in \
                 choices_sorted_by_num:
             all_menu.Append(choiceid, "filler")
@@ -137,11 +131,6 @@ class NameSubscriberComboBox(wx.Panel):
                     in sorted_submenus])
         self.PopupMenu(menu)
         menu.Destroy()
-
-    def menu_selected(self, evt):
-        self.combo_dlg.Selection = self.IDs.index(evt.Id)
-        # fake a choice
-        self.choice_made(evt)
 
     def GetItems(self):
         return self.orig_choices

